tests/unit/Sync_test.cpp

```diff
@@ -1330,44 +1303,1456 @@ class MockDbAccess : public mega::DbAccess
+
+    // Slots should be empty.
+    EXPECT_TRUE(slots.empty());
 }
 
-TEST(Sync, SyncConfigBag_withPreviousState)
+TEST_F(JSONSyncConfigIOContextTest, GetSlotsOrderedByModificationTime)
 {
-    std::vector<std::pair<uint32_t, std::string>> mData;
-    MockDbAccess dbaccess{mData};
-    mt::DefaultedFileSystemAccess fsaccess;
-    mega::PrnGen rng;
+    const size_t NUM_SLOTS = 3;
+
+    // Make sure drive path exists.
+    Directory drive(fsAccess(), Utilities::randomPath());
+
+    // Generate some slots for this user.
+    {
+        LocalPath configPath = drive;
+
+        // Generate suitable config path prefix.
+        configPath.appendWithSeparator(
+          LocalPath::fromPath(configName(), fsAccess()), false);
+
+        for (size_t i = 0; i < NUM_SLOTS; ++i)
+        {
+            using std::to_string;
+
+            ScopedLengthRestore restorer(configPath);
+
+            // Generate suffix.
+            LocalPath suffixPath =
+              LocalPath::fromPath("." + to_string(i), fsAccess());
+
+            // Complete config path.
+            configPath.append(suffixPath);
+
+            // Populate the file.
+            EXPECT_TRUE(Utilities::randomFile(configPath));
+
+            // Set the modification time.
+            EXPECT_TRUE(fsAccess().setmtimelocal(configPath, i * 1000));
+        }
+    }
+
+    vector<unsigned int> slots;
+
+    // Get the slots.
+    EXPECT_EQ(ioContext().getSlotsInOrder(drive.path(), slots), API_OK);
 
-    mega::SyncConfigBag bag1{dbaccess, fsaccess, rng, "some_id"};
-    mega::SyncConfig config1{"foo", "foo", 41, "remote", 122, {}, true, mega::SyncConfig::Type::TYPE_TWOWAY, false, true, mega::LOCAL_FINGERPRINT_MISMATCH};
-    config1.setBackupId(12345);
-    bag1.insert(config1);
-    ASSERT_EQ(1u, mData.size());
-    mega::SyncConfig config2{"bar", "bar", 42, "remote", 123, {}, false, mega::SyncConfig::Type::TYPE_UP, true, false, mega::NO_SYNC_ERROR};
-    config2.setBackupId(123456);
-    bag1.insert(config2);
-    ASSERT_EQ(2u, mData.size());
-    mega::SyncConfig config3{"bar2", "bar2", 43, "remote", 124, {}, false, mega::SyncConfig::Type::TYPE_UP, true, false, mega::NO_SYNC_ERROR};
-    config3.setBackupId(1234567);
-    bag1.insert(config3);
-    ASSERT_EQ(3u, mData.size());
-    bag1.insert(config3); // update
-    ASSERT_EQ(3u, mData.size());
-    bag1.removeByBackupId(1234567);
-    ASSERT_EQ(2u, mData.size());
+    // Did we retrieve the correct number of slots?
+    ASSERT_EQ(slots.size(), NUM_SLOTS);
 
-    const mega::SyncConfigBag bag2{dbaccess, fsaccess, rng, "some_id"};
-    const std::vector<mega::SyncConfig> expConfigs{config1, config2};
-    //ASSERT_EQ(expConfigs, bag2.all());
+    // Are the slots ordered by descending modification time?
+    {
+        vector<unsigned int> expected(NUM_SLOTS, 0);
+
+        iota(begin(expected), end(expected), 0);
+
+        EXPECT_TRUE(equal(begin(expected), end(expected), rbegin(slots)));
```

Comment:
It fails to build on my machine:
```
In file included from /home/sergio/bin/gtest/googletest/include/gtest/gtest.h:62:0,
                 from /usr/local/include/gmock/internal/gmock-internal-utils.h:47,
                 from /usr/local/include/gmock/gmock-actions.h:51,
                 from /usr/local/include/gmock/gmock.h:59,
                 from ../../../../tests/unit/Sync_test.cpp:22:
../../../../tests/unit/Sync_test.cpp: In member function ‘virtual void JSONSyncConfigTests::JSONSyncConfigIOContextTest_GetSlotsOrderedBySlotSuffix_Test::TestBody()’:
../../../../tests/unit/Sync_test.cpp:1739:59: error: ‘rbegin’ was not declared in this scope
         EXPECT_TRUE(equal(begin(expected), end(expected), rbegin(slots)));
                                                           ^
```
Jenkins is complaining for all tasks now, but it seems to be related to the recent changes in develop for `MegaStringListPrivate`?

Reply:
I've substituted slots.rbegin() in place of rbegin(slots).

Hopefully this'll fix the issue :)

Reply:
Yeah, it works :)
Thanks